### ISSUE TITLE
Prompt user before session expiry

### ISSUE THREAD

**Issue by bobby-brennan** [kaltura/developer-platform#49]

**Comment by bobby-brennan** [comment]
I have a demo up here, where session length is reduced to 4 minutes (instead of 30):

http://18.237.78.121:3005

The user is prompted 2 minutes before expiration (i.e. after 28 minutes normally, but after 2 minutes in the demo). If they click "OK", the session is renewed (for another 30 minutes). If they click "Cancel", they are logged out immediately.

**Comment by jessp01** [comment]
Worked very well for me:)
Tested both flows.

**Comment by jessp01** [comment]
Hi @bobby-brennan - like I said, works well for me so as far as I'm concerned, can be closed and moved to the "Ready for testing" queue. However, I see this is controlled here:
https://github.com/kaltura/developer-platform/blob/master/javascript/auth.js#L3

I feel it would be nice to read the value from an ENV var so that it could be easily controlled w/o changing the code.

**Comment by bobby-brennan** [comment]
Agreed this would be convenient, but there's unfortunately no easy way to plumb ENV variables into client JavaScript code...this file would need to become a template (breaking highlighting) and generate a second JS file. I don't think this is worth the technical overhead

**Comment by jessp01** [comment]
Yeah, I suppose you're right:) OK, so lets just set the session duration to 60 minutes and modify the message as per yesterday's email discussion.

**Comment by bobby-brennan** [comment]
Increased here: c9431c6d743e42398b02763b0e4a13f1c8cef749

Switch to a UI-driven modal here: e453efb98aa3dd3759dd7de0334c31e7bc2e61ab

**Comment by jessp01** [comment]
Hi @bobby-brennan - I thought we decided to stick with the confirm() in our last discussion:
```
Subject: developer.k
Sent: Tuesday, May 15, 2018
```
Was there an additional exchange I wasn't copied on?

**Comment by jessp01** [comment]
Hi @bobby-brennan - I was also disturbed by confirm(), at least in Chrome. In Firefox the implementation is different and it doesn't bring the browser's window into focus when confirm() is triggered. 
https://github.com/kaltura/developer-platform/commit/e453efb98aa3dd3759dd7de0334c31e7bc2e61ab also enables us to use custom button labels which is good. 

However, as you said yourself on the thread:
>  I agree it's annoying, but if we change it to a UI prompt, the user will come back to the tab after an
> hour, click "OK", and the session will have already expired.

This is the current behaviour and I'm not so sure that's a good thing...

**Comment by bobby-brennan** [comment]
I set it up so that if the user clicks OK and the session expired, it'll
bring up the login prompt. I think that mitigates the issue a bit

**Comment by jessp01** [comment]
Yup, that was the behaviour I observed as well. I am OK with it if @zoharbabin is.
